File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectModel/src/main/java/ro/mpp2025/javaprojectui/Proba.java
package ro.mpp2025.javaprojectui;

import java.util.Objects;

public class Proba extends Entity<Integer> {
    private String tip;
    private CategorieVarsta varsta;

    public Proba(){}
    public Proba(String tip, CategorieVarsta varsta) {
        this.tip = tip;
        this.varsta = varsta;
    }

    public String getTip() {
        return tip;
    }

    public CategorieVarsta getVarsta() {
        return varsta;
    }

    public void setTip(String tip) {
        this.tip = tip;
    }

    public void setVarsta(CategorieVarsta varsta) {
        this.varsta = varsta;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Proba proba = (Proba) o;
        return Objects.equals(tip, proba.tip) && Objects.equals(varsta, proba.varsta);
    }

    @Override
    public int hashCode() {
        return Objects.hash(tip, varsta);
    }

    @Override
    public String toString() {
        return "Proba{" +
                "tip='" + tip + '\'' +
                ", varsta=" + varsta +
                '}';
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Model/Proba.cs
using System.Text.Json.Serialization;

namespace Model;
[Serializable]
public class Proba : Entity<int>
{
    private string tip;
    private CategorieVarsta varsta;
        
    public Proba(){}
    [JsonConstructor]
    public Proba(string tip, CategorieVarsta varsta)
    {
        this.tip = tip;
        this.varsta = varsta;
    }

    public override string ToString()
    {
        return $"{nameof(tip)}: {tip}, {nameof(varsta)}: {varsta}, {nameof(Tip)}: {Tip}, {nameof(Varsta)}: {Varsta}";
    }

    public string Tip
    {
        get { return tip; }
        set { tip = value; }
    }
        
    public CategorieVarsta Varsta
    {
        get { return varsta; }
        set { varsta = value; }
    }
        
    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;
            
        Proba proba = (Proba)obj;
        return Equals(tip, proba.tip) && Equals(varsta, proba.varsta);
    }
        
    public override int GetHashCode()
    {
        return HashCode.Combine(tip, varsta);
    }
}